### ISSUE TITLE
stress test UBSAN verification error

### ISSUE THREAD

**Issue by Yuval-Ariel** [speedb-io/speedb#557]
when running COMPILE_WITH_UBSAN=1 make -j16 blackbox_crash_test on 72d418163bea1fe01231cda127b2018686eaccb0 (version candidate)  the following error happens:
```
Iterator has key 0000000000003C59000000000000000900000000000000E57878787878, but expected state does not.
Column family: default, op_logs: S 0000000000003C59000000000000000900000000000000E27878 NNN SFP 0000000000003C59000000000000000900000000000000E778787878787878 PPP S 0000000000003C59000000000000000900000000000000E57878787878 
Verification failed :(
```

UBSAN hasnt been used since #16  so it might not be a new issue. the error does not reproduce with ASAN atm.

**Comment by Yuval-Ariel** [comment]
also when running the narrow crash test and UBSAN:
``` ***options/options_helper.cc:1128:43: runtime error: load of value 64, which is not a valid value for type 'bool'*** ```
cmd: 
```
db_stress with pid=43513: ./db_stress --acquire_snapshot_one_in=12500 --allow_concurrent_memtable_write=True --bloom_bits=1 --clear_column_family_one_in=0 --compaction_ttl=0 --compare_full_db_state_snapshot=0 --compression_zstd_max_train_bytes=0 --continuous_verification_interval=0 --customopspercent=2 --db=/tmp/rocksdb_crashtest_narrowzd_tss6j --delpercent=10 --delrangepercent=2 --destroy_db_initially=0 --enable_compaction_filter=0 --expected_values_dir=/tmp/rocksdb_crashtest_expected__ywxxy9_ --fail_if_options_file_error=True --get_current_wal_file_one_in=0 --get_live_files_one_in=50000 --get_sorted_wal_files_one_in=0 --iterpercent=4 --log2_keys_per_lock=1 --max_key_len=8 --memtablerep=speedb.HashSpdRepFactory --mmap_read=0 --nooverwritepercent=95 --ops_per_thread=50000 --partition_filters=0 --periodic_compaction_seconds=0 --prefix_size=-1 --prefixpercent=0 --progress_reports=0 --readpercent=51 --reopen=2 --ribbon_starting_level=0 --secondary_cache_uri= --seed=2292430251 --sst_file_manager_bytes_per_truncate=0 --sync_wal_one_in=25000 --use_direct_io_for_flush_and_compaction=0 --use_direct_reads=0 --use_merge=1 --use_multiget=0 --user_timestamp_size=0 --value_size_mult=8 --verify_db_one_in=50000 --writepercent=31
```

**Comment by Yuval-Ariel** [comment]
no errors are found when running with skiplist. need to recheck once #570 is done.

**Comment by Yuval-Ariel** [comment]
reran on current main (964a9b490a0b0f4e24fc17fb0e9f57b0a1b74733) with the fix for #570  and the issue seems resolved
